src/odemis/acq/stream/_base.py

```diff
@@ -861,6 +1086,10 @@ def _onNewData(self, dataflow, data):
         if not self.raw:
             self.raw.append(data)
         else:
-            self.raw[0] = data
+            if isinstance(self.raw, list):
+                self.raw[0] = data
+            else:
+                # TODO support AcquisitionData
```

Comment:
I removed the TODO. There's something that should be done here when self.raw is a DataArrayShadow?